src/compiler/emitter.ts

```diff
@@ -3052,6 +3054,17 @@ module ts {
                     write(": ");
                     // Even though this is stored as identifier treat it as an expression
                     // Short-hand, { x }, is equivalent of normal form { x: x }
+
+                    if (languageVersion < ScriptTarget.ES6) {

```

Comment:
Feels weird to have this nested condition that matches one disjunct of the outer condition. I guess instead I would prefer:
```ts
if (languageVersion < ScriptTarget.ES6) {
    // Emit the colon and the generated name
}
else if (resolver.getExpressionNameSubstitution(node.name)) {
    // Just emit the colon
}
```